diff --git a/pappl/system-accessors.c b/pappl/system-accessors.c
index fa51f5a..d1c27f3 100644
--- a/pappl/system-accessors.c
+++ b/pappl/system-accessors.c
@@ -1430,7 +1430,8 @@ papplSystemSetHostname(
       _pappl_dns_sd_t	master = _papplDNSSDInit(system);
 					  // DNS-SD master reference
 
-      avahi_client_set_host_name(master, value);
+      if (master)
+        avahi_client_set_host_name(master, value);
 #endif // HAVE_AVAHI
 
       sethostname(value, (int)strlen(value));


